src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs

```diff
@@ -265,18 +265,72 @@ class AdoConnector {
         }
 
         public async Async.Task Process((string, string)[] notificationInfo) {
-            var seen = false;
-            await foreach (var workItem in ExistingWorkItems()) {
-                await UpdateExisting(workItem, notificationInfo);
-                seen = true;
+            var matchingWorkItems = ExistingWorkItems();
+
+            var nonDuplicate = matchingWorkItems
+                .Where(wi => !IsDuplicate(wi));
+
+            var numNonDuplicate = await nonDuplicate.CountAsync();
+
+            if (numNonDuplicate > 1) {
+                var nonDuplicateWorkItemIds = nonDuplicate
+                    .Select(wi => wi.Id)
+                    .ToEnumerable();
+
+                var matchingWorkItemIds = matchingWorkItems
+                    .Select(wi => wi.Id)
+                    .ToEnumerable();
+
+                var extraTags = new List<(string, string)> {
+                    ("NonDuplicateWorkItemIds", JsonSerializer.Serialize(nonDuplicateWorkItemIds)),
+                    ("MatchingWorkItemIds", JsonSerializer.Serialize(matchingWorkItemIds))
+                };
+                extraTags.AddRange(notificationInfo);
+
+                _logTracer.WithTags(extraTags).Info($"Found more than 1 matching, non-duplicate work item");
+                await foreach (var workItem in nonDuplicate) {
+                    await UpdateExisting(workItem, notificationInfo);
+                }
+            } else if (numNonDuplicate == 1) {
+                await UpdateExisting(await nonDuplicate.SingleAsync(), notificationInfo);
             }
+              // We have matching work items but all are duplicates
+              else if (await matchingWorkItems.AnyAsync()) {
+                _logTracer.WithTags(notificationInfo).Info($"All matching work items were duplicates, re-opening the oldest one");
+                var oldestWorkItem = await matchingWorkItems.OrderBy(wi => wi.Id).FirstAsync();
+                await UpdateExisting(oldestWorkItem, notificationInfo);
 
-            if (!seen) {
+                _ = await _client.AddCommentAsync(
+                    new CommentCreate() {
+                        Text = "This work item was re-opened because OneFuzz could only find related work items that are marked as duplicate."
+                    },
+                    _project,
+                    (int)oldestWorkItem.Id!);
+            }
+              // We never saw a work item like this before, it must be new
+              else {
                 var entry = await CreateNew();
                 var adoEventType = "AdoNewItem";
                 _logTracer.WithTags(notificationInfo).Event($"{adoEventType} {entry.Id:Tag:WorkItemId}");
-
             }
         }
+
+        private static bool IsDuplicate(WorkItem wi) {
```

Comment:
I suggest renaming this to something like `ADODuplicate` (and the associated variables) to avoid confusion with what we call duplicate in the template. Ideally we should change the template itself to rename `onDuplicate` to `onExisting` but that would cause some backward compat issues.